packages/acceptance-tests/pkg-tests-specs/sources/dragon.test.js

```diff
@@ -580,9 +579,10 @@ describe(`Dragon tests`, () => {
           await expect(run(`install`)).resolves.toBeTruthy();
 
           // Make sure that both the root and dragon-test-11-b have access to the same instance.
-          // This is only possible with the PnP and pnpm linkers, because the node-modules linker
-          // can't fulfill the peer dependency promise. For the NM linker we test that it at least
-          // fulfills the require promise (installing dragon-test-11-a both under the aliased and original name).
+          // This is only possible with the PnP and pnpm linkers.
+          // The node-modules linker can't fullfil these requirements for aliased packages,
+          // without resorting to symlinks and a layout resembling pnpm for aliased dependencies,
+          // which will be too different from the classic node_modules layout for all the other dependencies.
```

Comment:
The symlink layout you described on Discord would only affect aliased dependencies, right? Given that they are fairly rare, that nothing would change for regular dependencies, and that it doesn't have the cyclic symlink issue, I feel like it might be worth a try - but your call.

Reply:
Yes. It should work. However, it needs to be implemented carefully. Plus it might be worth resorting to symlinks only when the dependency is actually peer dependend upon or when self references support is enabled. So, if we making this carefully, it's not trivial, need 2-3 days to spend for implementation and tests... Should be done in a separate PR I think.